tested/dsl/dsl_errors.py

```diff
@@ -0,0 +1,80 @@
+import sys
+import textwrap
+
+import yaml
+from jsonschema.exceptions import ValidationError
+
+class InvalidYamlError(ValueError):
+    pass
+
+class DslValidationError(ValueError):
+    pass
+
+def convert_validation_error_to_group(
+    error: ValidationError,
+) -> ExceptionGroup | Exception:
+    if not error.context and not error.cause:
+        if len(error.message) > 150:
+            message = error.message.replace(str(error.instance), "<DSL>")
+            note = "With <DSL> being: " + textwrap.shorten(str(error.instance), 500)
+        else:
+            message = error.message
+            note = None
+        converted = DslValidationError(
+            f"Validation error at {error.json_path}: " + message
+        )
+        if note:
+            converted.add_note(note)
+        return converted
+    elif error.cause:
+        return error.cause
+    elif error.context:
+        causes = [convert_validation_error_to_group(x) for x in error.context]
+        message = f"Validation error at {error.json_path}, caused by a sub-exception."
+        return ExceptionGroup(message, causes)
+    else:
+        return error
+
+def print_dsl_validation_errors(errors: list):
```

Comment:
I would replace `print` with `handle` or `raise`

Nothing is directly printed. Errors with extra help text are raised.
The printing happens at a later point in time.